src/codec/mod.rs

```diff
@@ -0,0 +1,160 @@
+use std::{io, mem, fmt, error};
+use std::io::{Read, Write};
+
+use net::{MAX_MESSAGE_LEN};
```

Comment:
Could MAX_MESSAGE_LEN be defined in this module, instead of defining it in `net`?